### PR TITLE
honk: init at 0.9.8

### PR DESCRIPTION
**Pull request by Huy-Ngo**
###### Description of changes

Honk is a minimal ActivityPub server.
https://humungus.tedunangst.com/r/honk

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by raphaelr — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -14339,6 +14339,8 @@ with pkgs;
   haxePackages = recurseIntoAttrs (callPackage ./haxe-packages.nix { });
   inherit (haxePackages) hxcpp;
 
+  honk = callPackage ../servers/honk { };
```
[the remark this answers is not in the document]
Comment (Huy-Ngo) [reply pkgs/top-level/all-packages.nix:14342]: ah darn, i didn't notice the headings

File: pkgs/servers/honk/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  buildInputs = [ sqlite ];
+  subPackages = [ "." ];
+
+  postInstall = ''
+    mkdir -p $out/share/${pname}
+    mkdir -p $out/share/doc/${pname}
+    mkdir -p $out/share/man/man{1,3,5,7,8}
+
+    ls docs
```
Comment (raphaelr) [inline pkgs/servers/honk/default.nix:21]:
```suggestion
```

**Review by ncfavier — COMMENTED** [review]
The man page installation could be simplified with [installShellFiles](https://nixos.org/manual/nixpkgs/unstable/#installshellfiles)

**Comment by Huy-Ngo** [comment]
Regarding the views folder: currently, I put it at $out/share/honk, but by default honk would check the current folder instead. Should I patch that to read from `/run/current-system/sw/share/honk` (likely not a good idea because the installation is probably not system-wide) or patch the man page to tell them where to find it?

**Comment by McSinyx** [comment]
You should replace it with $out/share/honk.

**Comment by Huy-Ngo** [comment]
I haven't tried that, but i don't think $out is available at runtime, nor can go read it with that syntax

**Comment by ncfavier** [comment]
If we want to provide a default value for `-viewdir`, then something like

```
wrapProgram "$out/bin/honk" --add-flags "-viewdir $out/share/honk/views"
```

would be the way to go. However, I wonder if we really need/want to do this, or if this should be the responsibility of the NixOS module we will eventually want to add.

**Comment by raphaelr** [comment]
> However, I wonder if we really need/want to do this, or if this should be the responsibility of the NixOS module we will eventually want to add.

Agreed

With the wrapProgram approach, you can't override the viewdir arg anymore at runtime.

If you do want to provide a default, there are two options:
1. Use substituteInPlace in a hook
```nix
postPatch = ''
  substituteInPlace main.go \
    "var viewDir = \".\"' \
    "var viewDir = \"$out/share/honk\""
'';
```

2. Add a patch, and use the `substituteAll` function to modify the patch before applying it. A good example is in [sway](https://github.com/NixOS/nixpkgs/blob/8dd2c70b69a2d58c4d1dc27c180804efac6b3bf9/pkgs/applications/window-managers/sway/default.nix#L38-L40).

**Comment by raphaelr** [comment]
Now the `intro.1` `vim.3`, and `activitypub.7` manpages are installed without the honk- prefix. At least for `intro` and `vim`, that's not great.

**Comment by Huy-Ngo** [comment]
I guess I will go with the patch one then. The user should be able to change the parameter.

As for the man pages, it looks like there's no better way for these than manual copy and rename then.

**Review by raphaelr — APPROVED** [review]
looks good

**Review by ncfavier — COMMENTED** [review]
I would suggest renaming the man pages first (`mv docs/{,honk-}intro.1`) and then installing them with `installManPage`.

**Comment by Huy-Ngo** [comment]
makes sense

**Review by ncfavier — APPROVED** [review]

**Pull request merged**